### PR TITLE
sc-11674: Update native SDKs to 3.5.x

### PR DESCRIPTION
**Pull request by jrtibbetts**
[sc-11674: [Capacitor SDK] Update to iOS 3.5.3 & Android 3.5.2](https://app.shortcut.com/radarlabs/story/11674)

* Updates iOS RadarSDK to [`3.5.3`](https://github.com/radarlabs/radar-sdk-ios/releases/tag/3.5.3) & Android RadarSDK to [`3.5.2`](https://github.com/radarlabs/radar-sdk-android/releases/tag/3.5.2).
* Adds `[Radar]Plugin.sendEvent()`.
* Adds an optional `chainMetadata` parameter to `[Radar]Plugin.searchPlaces()`.

### CODE REVIEW

File: ios/Plugin.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -287,11 +287,13 @@
 				"${PODS_ROOT}/Target Support Files/Pods-PluginTests/Pods-PluginTests-frameworks.sh",
 				"${BUILT_PRODUCTS_DIR}/Capacitor/Capacitor.framework",
 				"${BUILT_PRODUCTS_DIR}/CapacitorCordova/Cordova.framework",
+				"${BUILT_PRODUCTS_DIR}/RadarSDK/RadarSDK.framework",
```
Comment (jrtibbetts) [inline ios/Plugin.xcodeproj/project.pbxproj:290]: I'm not sure why this and line 296 weren't there before, but they were automatically added.

File: ios/Plugin/generate_plugin_m.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env ruby
+
```
Comment (jrtibbetts) [inline ios/Plugin/generate_plugin_m.rb:2]: A script to generate the `Plugin.m` file from `Plugin.swift`.

**Comment by tjulien** [comment]
Can we add `scheduledArrivalAt` to `RadarTrip` in definitions.ts?

That's the critical thing needed to get out this week.  (Basically add support in Capacitor for https://github.com/radarlabs/radar-sdk-ios/pull/185).

If we can add that and test it end to end we should be good to ship this!

File: ios/Plugin/Plugin.swift [hunk 3]
```diff
@@ -595,4 +689,76 @@ public class RadarPlugin: CAPPlugin, RadarDelegate {
...
+}
+
+extension RadarTripOptions {
+
+    /// The native SDK doesn't yet set `scheduledArrivalAt` or
+    /// `approachingThreshold` from a dictionary, so do those here, for now.
+    static func tripOptionsFromJSDictionary(_ jsDictionary: [AnyHashable: Any]) -> RadarTripOptions {
+        let options = RadarTripOptions(from: jsDictionary)
+        options.scheduledArrivalAt = jsDictionary["scheduledArrivalAt"] as? Date
+        options.approachingThreshold = UInt8(jsDictionary["approachingThreshold"] as? Int ?? 0)
```
Comment (brettguenther) [inline ios/Plugin/Plugin.swift:759]: I am not familiar with this, but will this override the approaching threshold to 0? if so def don't want that/want to keep this null/unset

File: src/definitions.ts [hunk 4]
```diff
@@ -27,8 +27,9 @@ export interface RadarPlugin {
   cancelTrip(): Promise<RadarTripCallback>;
   acceptEvent(options: { eventId: string, verifiedPlaceId: string }): void;
   rejectEvent(options: { eventId: string }): void;
+  sendEvent(options: { customType: string, location?: { latitude: number, longitude: number }, metadata?: object }): Promise<RadarEventCallback>;
   getContext(options?: { latitude?: number, longitude?: number }): Promise<RadarContextCallback>;
-  searchPlaces(options: { near?: { latitude: number, longitude: number }, radius: number, chains?: string[], categories?: string[], groups?: string[], limit: number }): Promise<RadarSearchPlacesCallback>;
+  searchPlaces(options: { near?: { latitude: number, longitude: number }, radius: number, chains?: string[], chainsMetadata?: object, categories?: string[], groups?: string[], limit: number }): Promise<RadarSearchPlacesCallback>;
```
[the remark this answers is not in the document]
Comment (jrtibbetts) [reply src/definitions.ts:32]: Oof. Good catch.

File: ios/Plugin/Plugin.swift (hunk 3 above)
Comment (jrtibbetts) [reply ios/Plugin/Plugin.swift:759]: Yes, you're right.

File: src/definitions.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (jrtibbetts) [reply src/definitions.ts]: Fixed in commit #15d960.

File: ios/Plugin/Plugin.swift (hunk 3 above)
Comment (jrtibbetts) [reply ios/Plugin/Plugin.swift]: Fixed in commit #15d960.

**Pull request closed without merging**